src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -525,6 +534,13 @@ private static unsafe string FormatDouble(ref ValueStringBuilder sb, double valu
             char fmt = ParseFormatSpecifier(format, out int precision);
             byte* pDigits = stackalloc byte[DoubleNumberBufferLength];
 
+            if (fmt == '\0')
```

Comment:
(super nit): I think we should be consistent with how we are checking the `fmt` variable in this method.

Here we are using `fmt == '\0'`.  Below: `fmt != 0`.

Reply:
I can update the below one in a separate PR.

I think explicitly checking `'\0'` makes it much clearer that we are checking for the terminating null.